### PR TITLE
Feature - Create more detailed error responses when submitting batch jobs

### LINKED ISSUE #1748: Better Error Message When Rejecting Jobs
When we reject a job (for example because it can't be scheduled on any node) the whole grpc call fails.  There is an error message that is returned, but users have said that this doesn't give enough infomation about what was valid. Instead they ask:

```
is it possible for these errors to be returned on the Error field on the JobResponseItem that we're given?
```

┆Issue is synchronized with this [Jira Task](https://gr-oss.atlassian.net/browse/BATCH-80) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by Raajheer1**
#### What type of PR is this?
This PR adds functionality to JobSubmitResponse when an error has occured. Previously the only thing returned was the error. The changes made in this PR allow the code to do 2 things:
- Build a collection of errors, if mulitple jobs error for the same return, these changes will return them at the same time to the user. Rather than them having to reattempt submissions over and over again with 1 fix each time.
- Return more detailed errors to the user by utilizing the JobSubmitResponse Struct

#### Which issue(s) this PR fixes:
Fixed #1748 

┆Issue is synchronized with this [Jira Task](https://gr-oss.atlassian.net/browse/BATCH-412) by [Unito](https://www.unito.io)

### CODE REVIEW

File: internal/armada/server/lease.go [hunk 1]
```diff
@@ -913,11 +913,11 @@ func (q *AggregatedQueueServer) addAvoidNodeAffinity(
 		}
 
 		changed := addAvoidNodeAffinity(jobs[0], labels, func(jobsToValidate []*api.Job) error {
-			if ok, err := validateJobsCanBeScheduled(jobsToValidate, allClusterSchedulingInfo); !ok {
+			if ok, responseItems, err := validateJobsCanBeScheduled(jobsToValidate, allClusterSchedulingInfo); !ok {
 				if err != nil {
-					return errors.WithMessage(err, "can't schedule at least 1 job")
+					return errors.WithMessagef(err, "can't schedule %d job", len(responseItems))
```
Comment (richscott) [inline internal/armada/server/lease.go:918]: I'd change `job` to `job(s)`.

**Review by richscott — COMMENTED** [review]
I think it looks good so far! I had just one suggestion re an error message spelling. Also, I think you'll want to add some tests, of course.

**Review by richscott — DISMISSED** [review]

**Comment by Raajheer1** [comment]
Armada Before though armadactl
Armada-Before

Armada After through armadactl
Armada-After

Submitted a bad sleep yaml job containing two jobs, both have invalid podSpecs as the requested and limit resources both contain "memory: 0"

**Review by richscott — APPROVED** [review]

**Comment by richscott** [comment]
This changes in this PR were merged into master a couple weeks ago, via a separate PR - closing this.

**Pull request closed without merging**